lib/llvmopencl/LLVMUtils.cc

```diff
@@ -726,10 +726,16 @@ llvm::Type *SizeT(llvm::Module *M) {
   return IntegerType::get(M->getContext(), AddressBits);
 }
 
-bool isWorkitemFunctionWithOnlyCompilerExpandableCalls(const llvm::Function &F) {
-  Instruction::const_use_iterator UI = F.use_begin(), UE = F.use_end();
-  for (; UI != UE; ++UI) {
-    llvm::CallInst *Call = dyn_cast<llvm::CallInst>(UI->getUser());
+bool isWorkitemFunctionWithOnlyCompilerExpandableCalls(
+    const llvm::Function &F) {
+
+  if (F.getName() != GID_BUILTIN_NAME && F.getName() != GS_BUILTIN_NAME &&
+      F.getName() != GROUP_ID_BUILTIN_NAME && F.getName() != LID_BUILTIN_NAME &&
+      F.getName() != LS_BUILTIN_NAME && F.getName() != NGROUPS_BUILTIN_NAME)
+    return false;
```

Comment:
My two cents:
```suggestion
  if (!CompilerExpandableCalls.count(F.getName()))
    return false;
```
Where the `CompilerExpandableCalls` is a static object of a set.

Reply:
Yep, not in my taste :)